lib/cosmos/utilities/sleeper.rb

```diff
@@ -0,0 +1,37 @@
+# encoding: ascii-8bit
+
+# Copyright 2014 Ball Aerospace & Technologies Corp.
+# All Rights Reserved.
+#
+# This program is free software; you can modify and/or redistribute it
+# under the terms of the GNU General Public License
+# as published by the Free Software Foundation; version 3 with
+# attribution addendums as found in the LICENSE.txt
+
+module Cosmos
+
+  class Sleeper
```

Comment:
I just read http://www.sitepoint.com/the-self-pipe-trick-explained/ and think I have it figured out. The sleep routine is really sleep_and_check_for_cancel all rolled into one. Maybe just a note about this being the self-pipe trick and how sleep is being using with the return values.